Design/DesignDoc.md

```diff
@@ -0,0 +1,220 @@
+
+* Adding, updating, and retrieving/viewing expenditure information
+* User specific data (multiple users allowed)
+* Retrieving expenditure information based on tags
+
+### 4.2. Out of Scope
+
+_Based on your problem description in Sections 1 and 2, are there any aspects you are not planning to solve? Do potential expansions or related problems occur to you that you want to explicitly say you are not worrying about now? Feel free to put anything here that you think your team can't accomplish in the unit, but would love to do with more time._
+
+* Displaying expenditure and budget information in a graphical representation 
+* Rolling over remaining budget information for the next term 
+* Calculating balance remaining in budget after expenses  
+
+
+
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+This initial iteration will provide the minimum viable product (MVP) including adding, retrieving, and updating an expense information for the user role.
+
+We will use API Gateway and Lambda to create 6 endpoints (GetExpense, Add/UpdateExpense, GetTaggedExpense, GetBudget, Add/UpdateBudget, BudgetStatus) that will handle the creation, update, and retrieval of expense and budget information to satisfy our requirements.
+
+We will store expense and budget information in a table in DynamoDB.
+
+Amazon Elite Budget Service will also provide a web interface for users. A main page providing a variety of options to add, update and view expense and budget information.
+
+
+
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ExpenseModel
+
+String expenseId;
+String expenseName;
+String expenseAmount; 
+String tag;
+ZonedDateTime expenseDate;
+
+// BudgetModel
+
+String budgetId;
+String targetAmount; 
+Boolean status; 
+ZonedDateTime date;
+
+```
+
+## 6.2. _GetExpense Endpoint_
+
+* Accepts `GET` requests to `/expense/:date`
+* Accepts an expenseID and returns the corresponding ExpenseModel.
+    * If the given expenseID is not found, will throw a
+      `ExpenditureNotFoundException`
+
+## 6.3 _GetAllExpenses Endpoint_
+
+* Accepts `GET` requests to `/expense`
+* Returns all the expenses with any tag in the ExpenseModel format.
+    * If there is no data found, will throw a
+      `NoDataFoundException`
+
+## 6.4. GetTaggedExpense Endpoint
+
+* Accepts `GET` requests to `/expense/:tag`
```

Comment:
You'll need to add something to this URL to make it unique from the GetExpense endpoint. E.g. maybe `/expense/tags/:tag`